master/internal/sproto/task.go

```diff
@@ -65,3 +65,33 @@ type Allocation interface {
 	Start(ctx *actor.Context, spec tasks.TaskSpec)
 	Kill(ctx *actor.Context)
 }
+
+type (
+	// ValidateCommandResourcesRequest is a message asking resource manager whether the given
+	// resource pool can (or, rather, if it's not impossible to) fulfill the command request
+	// for the given amount of slots.
+	ValidateCommandResourcesRequest struct {
+		ResourcePool string
+		Slots        int
+	}
+
+	// ValidateCommandResourcesResponse is the response to ValidateCommandResourcesRequest.
+	ValidateCommandResourcesResponse struct {
+		// Fulfillable values:
+		// - nil: unknown
+		// - false: impossible to fulfill
+		// - true: ok
+		Fulfillable *bool
```

Comment:
I would prefer if this is just a `bool` not `*bool`. Do we really need unknown? On prem agents are still within a resource pool afaik and have slots per agent -1. I don't see any case where we could have a nil `*bool` in the code.